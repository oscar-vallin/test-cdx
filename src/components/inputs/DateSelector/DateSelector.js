import { addMonths, addYears } from '@fluentui/date-time-utilities';
import { DatePicker, DayOfWeek } from 'office-ui-fabric-react/lib-commonjs/DatePicker';

const today = new Date(Date.now());
const minDate = addMonths(today, -1);
const maxDate = addYears(today, 1);

const DayPickerStrings = {
  months: [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ],

  shortMonths: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],

  days: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],

  shortDays: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],

  goToToday: 'Go to today',
  prevMonthAriaLabel: 'Go to previous month',
  nextMonthAriaLabel: 'Go to next month',
  prevYearAriaLabel: 'Go to previous year',
  nextYearAriaLabel: 'Go to next year',
  closeButtonAriaLabel: 'Close date picker',
  monthPickerHeaderAriaLabel: '{0}, select to change the year',
  yearPickerHeaderAriaLabel: '{0}, select to change the month',

  isRequiredErrorMessage: 'Field is required.',

  invalidInputErrorMessage: 'Invalid date format.',

  isOutOfBoundsErrorMessage: `Date must be between ${minDate.toLocaleDateString()}-${maxDate.toLocaleDateString()}`,
};

const firstDayOfWeek = DayOfWeek.Sunday;

export const DateSelector = ({ id = '', onChange, required }) => {
  return (
    <DatePicker
      id={id}
      isRequired={required}
      firstDayOfWeek={firstDayOfWeek}
      strings={DayPickerStrings}
      minDate={minDate}
      maxDate={maxDate}
      onSelectDate={onChange}
      initialPickerDate={today}
      allowTextInput
    >
      AAAAAAAA
    </DatePicker>
  );
};

export default DateSelector;
