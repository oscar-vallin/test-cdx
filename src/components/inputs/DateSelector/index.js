import { DateSelector } from './DateSelector';

export { DateSelector };
