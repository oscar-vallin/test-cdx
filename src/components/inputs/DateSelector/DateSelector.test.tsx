import React from 'react';
import ReactDOM from 'react-dom';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { DayOfWeek } from 'office-ui-fabric-react/lib-commonjs/DatePicker';
import { fireEvent, render, screen } from '@testing-library/react';
import { addDays, addMonths, addYears } from '@fluentui/date-time-utilities';
import { getHours } from 'date-fns';
import { DateSelector as Component } from './DateSelector.js';

const today = new Date();
const yesterday = addDays(today, -1);
const hour = getHours(new Date());
const firstDayOfWeek = DayOfWeek.Sunday;
const minDate = addMonths(today, -2);
const maxDate = addYears(today, 1);

const placeholderText = 'This is a placeholder';

const defaultProps = {
  id: 'InputDate',
  Label: 'date',
  value: hour < 9 ? yesterday : today,
  required: false,
  onChange: (e: React.FormEvent<HTMLInputElement>) => {
    return e?.currentTarget?.value ?? '';
  },
};

test('Matches Snapshot', () => {
  const wrapper = shallow(<Component {...defaultProps} placeholder={placeholderText} />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

describe('Basic InputDate Component', () => {
  it('Should renders InputDate Component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Component {...defaultProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should renders InputDate with placeholder', () => {
    const wrapper = shallow(<Component {...defaultProps} placeholder={placeholderText} />);
    expect(wrapper.prop('placeholder')).toEqual(placeholderText);
  });

  it('Should renders InputDate with Props', () => {
    const wrapper = shallow(<Component {...defaultProps} placeholder={placeholderText} />);
    expect(wrapper.prop('value')).toEqual(defaultProps.value);
    expect(wrapper.prop('required')).toBeFalsy();
    expect(wrapper.prop('placeholder')).toEqual(placeholderText);
  });

  it('Should have the first day of the week as parameter', () => {
    const wrapper = shallow(<Component {...defaultProps} placeholder={placeholderText} />);
    expect(wrapper.prop('firstDayOfWeek')).toEqual(firstDayOfWeek);
  });

  it('@Testing: Render Input', () => {
    render(<Component {...defaultProps} placeholder={placeholderText} />);
    const input = screen.getByPlaceholderText(placeholderText);
    expect(input).toBeInTheDocument();
  });

  it('@Testing: Render by Role', () => {
    render(<Component {...defaultProps} placeholder={placeholderText} />);
    const input = screen.getByRole('combobox');
    expect(input).toBeInTheDocument();
  });

  it('@Testing: Input Text', () => {
    render(<Component {...defaultProps} placeholder={placeholderText} />);
    const input = screen.getByRole('combobox');
    fireEvent.click(input);
    expect(input).toHaveValue(today.toDateString());
  });
});
