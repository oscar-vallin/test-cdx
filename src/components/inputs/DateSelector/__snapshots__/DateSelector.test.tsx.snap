// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<StyledDatePickerBase
  allowTextInput={true}
  firstDayOfWeek={0}
  initialPickerDate={2021-08-31T22:59:52.484Z}
  isRequired={false}
  maxDate={2022-08-31T22:59:52.484Z}
  minDate={2021-07-31T22:59:52.484Z}
  onSelectDate={[Function]}
  strings={
    Object {
      "closeButtonAriaLabel": "Close date picker",
      "days": Array [
        "Sunday",
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
        "Saturday",
      ],
      "goToToday": "Go to today",
      "invalidInputErrorMessage": "Invalid date format.",
      "isOutOfBoundsErrorMessage": "Date must be between 7/31/2021-8/31/2022",
      "isRequiredErrorMessage": "Field is required.",
      "monthPickerHeaderAriaLabel": "{0}, select to change the year",
      "months": Array [
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December",
      ],
      "nextMonthAriaLabel": "Go to next month",
      "nextYearAriaLabel": "Go to next year",
      "prevMonthAriaLabel": "Go to previous month",
      "prevYearAriaLabel": "Go to previous year",
      "shortDays": Array [
        "S",
        "M",
        "T",
        "W",
        "T",
        "F",
        "S",
      ],
      "shortMonths": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "yearPickerHeaderAriaLabel": "{0}, select to change the month",
    }
  }
>
  AAAAAAAA
</StyledDatePickerBase>
`;
